### PR TITLE
MsGraph User with Common Api

### PR DESCRIPTION
**Pull request by Bargenish**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: demisto/etc#20893

## Description
MSGraph User to packs
Implementing MicrosoftApiModule
Added support to authenticate using a self-deployed Azure application.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 4.5.0
- [ ] 5.0.0
- [ ] 5.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation 

## Demisto Partner?
- [ ] The title must be in the following format: **[YOUR_PARTNER_ID] short description**

### CODE REVIEW

File: Packs/MicrosoftGraphUser/Integrations/MicrosoftGraphUser/README.md [hunk 1]
```diff
@@ -44,6 +44,16 @@
 <li>Click <strong>Test</strong> to validate the URLs, token, and connection.</li>
 </ol>
 </div>
+
+<h2>Use a self deployed Azure application</h2>
+<p>To use a self configured Azure application, a new Azure App Registration should be added in the Azure Portal. To add such registration, refer to the following Microsoft article:
+<a href="https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app">https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app</a></p>
+<p>The Tenant ID, Client ID and Client secret are needed for the integration.
+To configure the integration in Demisto to use the application, place those parameters in the following manner (instead of you received from the admin consent in the current doc):
```
Comment (Itay4) [inline Packs/MicrosoftGraphUser/Integrations/MicrosoftGraphUser/README.md:52]: please add here required permissions that this app need to have

File: Packs/MicrosoftGraphUser/Integrations/MicrosoftGraphUser/MicrosoftGraphUser.py [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+
+def main():
+    params: dict = demisto.params()
+    url = params.get('host', '').rstrip('/') + '/v1.0/'
+    tenant = params.get('tenant_id')
+    auth_and_token_url = params.get('auth_id', '')
+    enc_key = params.get('enc_key')
+    verify = not params.get('insecure', False)
+    self_deployed: bool = params.get('self_deployed', False)
+    # TODO: check it does the same as handle proxy
```
Comment (Itay4) [inline Packs/MicrosoftGraphUser/Integrations/MicrosoftGraphUser/MicrosoftGraphUser.py:250]: what about this TODO ?

Comment (Bargenish) [reply Packs/MicrosoftGraphUser/Integrations/MicrosoftGraphUser/MicrosoftGraphUser.py:250]: I checked it and forgot to remove this line. sorry.

**Comment by kirbles19** [comment]
@Bargenish Done.

**Comment by Bargenish** [comment]
@kirbles19 thanks!

**Pull request merged**